test/manual/tls_support.test.ts

```diff
@@ -47,36 +48,59 @@ describe('TLS Support', function () {
     });
 
     context('when tls filepaths have length > 0', () => {
-      beforeEach(async () => {
-        client = new MongoClient(CONNECTION_STRING, tlsSettings);
-      });
+      context('when connection will succeed', () => {
+        beforeEach(async () => {
+          client = new MongoClient(CONNECTION_STRING, tlsSettings);
+        });
+
+        it('should read in files async at connect time', async () => {
+          expect(client.options).property('tlsCAFile', TLS_CA_FILE);
+          expect(client.options).property('tlsCertificateKeyFile', TLS_CERT_KEY_FILE);
+          expect(client.options).not.have.property('ca');
+          expect(client.options).not.have.property('key');
+          expect(client.options).not.have.property('cert');
+
+          await client.connect();
+
+          expect(client.options).property('ca').to.exist;
+          expect(client.options).property('key').to.exist;
+          expect(client.options).property('cert').to.exist;
+        });
 
-      it('should read in files async at connect time', async () => {
-        expect(client.options).property('tlsCAFile', TLS_CA_FILE);
-        expect(client.options).property('tlsCertificateKeyFile', TLS_CERT_KEY_FILE);
-        expect(client.options).not.have.property('ca');
-        expect(client.options).not.have.property('key');
-        expect(client.options).not.have.property('cert');
+        context('when client has been opened and closed more than once', function () {
+          it('should only read files once', async () => {
+            await client.connect();
+            await client.close();
 
-        await client.connect();
+            const caFileAccessTime = (await fs.stat(TLS_CA_FILE)).atime;
+            const certKeyFileAccessTime = (await fs.stat(TLS_CERT_KEY_FILE)).atime;
 
-        expect(client.options).property('ca').to.exist;
-        expect(client.options).property('key').to.exist;
-        expect(client.options).property('cert').to.exist;
+            await client.connect();
+
+            expect((await fs.stat(TLS_CA_FILE)).atime).to.deep.equal(caFileAccessTime);
+            expect((await fs.stat(TLS_CERT_KEY_FILE)).atime).to.deep.equal(certKeyFileAccessTime);
```

Comment:
Can we add assertions similar to these for the CRL file?

Reply:
Done